src/plugins/mer/mergeneraloptionswidget.cpp

```diff
@@ -58,13 +66,111 @@ MerGeneralOptionsWidget::MerGeneralOptionsWidget(QWidget *parent)
     m_ui->benchLocationPathChooser->setExpectedKind(PathChooser::ExistingCommand);
     m_ui->benchLocationPathChooser->setPath(MerSettings::qmlLiveBenchLocation());
     m_ui->benchSyncWorkspaceCheckBox->setChecked(MerSettings::isSyncQmlLiveWorkspaceEnabled());
+
+    m_ui->dconfSettingsWidget->setSummaryFontBold(true);
+    m_ui->dconfSettingsWidget->setSummaryText(tr("Device dconf settings"));
+    m_ui->dconfSettingsWidget->setWidget(m_optionsWidget);
+
+    MerSettings::populateWithDeviceModels(m_ui->deviceModelComboBox);
+
+    connect(m_ui->screenResolutionHeightSpinBox, &QAbstractSpinBox::editingFinished,
+            this, &MerGeneralOptionsWidget::deviceModelScreenChanged);
+    connect(m_ui->screenResolutionWidthSpinBox, &QAbstractSpinBox::editingFinished,
+            this, &MerGeneralOptionsWidget::deviceModelScreenChanged);
+    connect(m_ui->screenHeightSpinBox, &QAbstractSpinBox::editingFinished,
+            this, &MerGeneralOptionsWidget::deviceModelScreenChanged);
+    connect(m_ui->screenWidthSpinBox, &QAbstractSpinBox::editingFinished,
+            this, &MerGeneralOptionsWidget::deviceModelScreenChanged);
+    connect(m_optionsWidget, SIGNAL(optionsChanged(QString)),
+            SLOT(deviceModelDconfChanged(QString)));
+    connect(m_ui->deviceModelComboBox, SIGNAL(currentIndexChanged(int)),
+            SLOT(updateGui()));
+    connect(MerSettings::instance(), &MerSettings::deviceModelChanged,
+            this, &MerGeneralOptionsWidget::onDeviceModelEdited);
+
+    updateGui();
 }
 
 MerGeneralOptionsWidget::~MerGeneralOptionsWidget()
 {
+    MerSettings::deviceModelsReset();
+
     delete m_ui;
 }
 
+void MerGeneralOptionsWidget::onDeviceModelEdited()
+{
+    MerSettings::populateWithDeviceModels(m_ui->deviceModelComboBox);
+}
+
+void MerGeneralOptionsWidget::deviceModelDconfChanged(const QString &value)
+{
+    const QString name = m_ui->deviceModelComboBox->currentData().toString();
+    const MerEmulatorDeviceModel model = MerSettings::deviceModels().value(name);
+    QTC_ASSERT(!model.isNull(), return);
+
+    if (model.dconf() != value) {
+        MerSettings::insertDeviceModel(MerEmulatorDeviceModel(name,
```

Comment:
"insert" is usually used when adding something at specified position. I would consider "addDeviceModel" instead.

Reply:
Name changed.